.github/workflows/cleardown-terraform-state.yaml

```diff
@@ -0,0 +1,45 @@
+name: Cleardown Terraform state
+
+on:
+  workflow_call:
+    inputs:
+      github_environment:
+        description: "Defines the Github environment in which to pull environment variables from"
+        required: true
+        type: string
+      deployment_workspace:
+        description: 'Name of the terraform workspace to tear down the service infrastructure'
+        required: false
+        default: 'not-set'
+        type: string
+
+jobs:
+  echo-workflow-inputs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: echo workflow inputs
+        run: |
+          echo "github environment: ${{ inputs.github_environment }}"
+          echo "deployment workspace: ${{ inputs.deployment_workspace }}"
+  cleardown-terraform-state:
+    runs-on: ubuntu-latest
+    environment: ${{ inputs.github_environment }}
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Checkout Repository
```

Comment:
To be consistent I think we need to use the wrapper action here /.github/actions/checkout-repository